root.go

```diff
@@ -69,6 +69,7 @@ type FSNode interface {
 	GetNode() (ipld.Node, error)
 	Flush() error
 	Type() NodeType
+	RemoveParent()
```

Comment:
Exposing `RemoveParent` on a public interface probably isn't a good idea (it's an internal function only meant to be called by `Unlink`. We could either:

1. Define a private interface.
2. Call it `Unlink()` and have files unlink themselves (the child would need access to the parent somehow).

Reply:
> * Define a private interface.

Yes, that sounds good, I'm abusing the `FSNode` interface here (because in my mental model `FSNode` just means child of a directory -either another dir or a file- but I'm thinking in OOP terms and I don't think I can have a public interface with private methods).